### ISSUE TITLE
Clean up ABDK Math Quad Imports

### ISSUE THREAD

**Issue by 0x4007** [pavlovcik/uad-ui#690]
I noticed that there are some imports as a submodule and others as a direct file in the codebase. We should only import from the submodule.

Working here now and seeing this new way to import them. 

https://github.com/pavlovcik/ubiquity-dollar/blob/3c5fee00aaaf142a649bbee961bac5d5ade15baf/packages/contracts/src/dollar/libraries/lib-credit-nft-redemption-calculator.sol#L5

(draft issue)

**Comment by rndquu** [comment]
> I noticed that there are some imports as a submodule and others as a direct file in the codebase. We should only import from the submodule.
> 
> Working here now and seeing this new way to import them.
> 
> https://github.com/pavlovcik/ubiquity-dollar/blob/3c5fee00aaaf142a649bbee961bac5d5ade15baf/packages/contracts/src/dollar/libraries/lib-credit-nft-redemption-calculator.sol#L5
> 
> (draft issue)

Could you show an example of a "direct file import" where submodule import should have been used?

**Comment by 0x4007** [comment]
I think its due to the remapping. Here's some search results related to abdk imports.

```
import "abdk/ABDKMathQuad.sol";
import "abdk/ABDKMathQuad.sol";
import "abdk/ABDKMathQuad.sol";
import "abdk-libraries-solidity/ABDKMathQuad.sol";
import "abdk-libraries-solidity/ABDKMathQuad.sol";
import "abdk/ABDKMathQuad.sol";
import "abdk-libraries-solidity/ABDKMathQuad.sol";
import "abdk-libraries-solidity/ABDKMathQuad.sol";
import "abdk/ABDKMathQuad.sol";
import "abdk/ABDKMathQuad.sol";
import "abdk-libraries-solidity/ABDKMathQuad.sol";
```

**Comment by 0x4007** [comment]
The auto assign is dope to see. Maybe we can even remove the assign command so that its only assigned when draft pull requests are opened.

**Comment by rndquu** [comment]
> I think its due to the remapping. Here's some search results related to abdk imports.
> 
> ```
> import "abdk/ABDKMathQuad.sol";
> import "abdk/ABDKMathQuad.sol";
> import "abdk/ABDKMathQuad.sol";
> import "abdk-libraries-solidity/ABDKMathQuad.sol";
> import "abdk-libraries-solidity/ABDKMathQuad.sol";
> import "abdk/ABDKMathQuad.sol";
> import "abdk-libraries-solidity/ABDKMathQuad.sol";
> import "abdk-libraries-solidity/ABDKMathQuad.sol";
> import "abdk/ABDKMathQuad.sol";
> import "abdk/ABDKMathQuad.sol";
> import "abdk-libraries-solidity/ABDKMathQuad.sol";
> ```

yes, if you run `forge remappings` you get 2 `abdk` remapping values:
```
abdk-libraries-solidity/=lib/abdk-libraries-solidity/
abdk/=lib/abdk-libraries-solidity/
```

But we have only 1 [defined](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/remappings.txt#L8) so it is not clear to me where the 2nd one `abdk-libraries-solidity/=lib/abdk-libraries-solidity/` is taken from, perhaps from some other git submodule

**Comment by rndquu** [comment]
> The auto assign is dope to see. Maybe we can even remove the assign command so that its only assigned when draft pull requests are opened.

When a new bounty hunter opens a repo where the bot is installed he checks closed issues and finds out the `/assign` command. This way he understands how to deal with the bot. Bounty hunters will spend more time deducting how to self assign an issue via a draft PR compared to the straight `/assign` command. Right now we use both methods and most of the issues are assigned via the `/assign` command.

**Comment by 0x4007** [comment]
> ### [ **[ CLAIM 25 WXDAI ]** ](https://pay.ubq.fi?claim=eyJwZXJtaXQiOnsicGVybWl0dGVkIjp7InRva2VuIjoiMHhlOTFEMTUzRTBiNDE1MThBMkNlOERkM0Q3OTQ0RmE4NjM0NjNhOTdkIiwiYW1vdW50IjoiMjUwMDAwMDAwMDAwMDAwMDAwMDAifSwibm9uY2UiOiIzMzU0NjMzMjc3NTExMTIzMTA5ODU5OTI4MjcyNTc5NDc0Mzg1MjI0OTU0NDQyMjAzOTE4NDc1NjAwNDY5MzE5MTc2MjA2OTg2Mzk3OSIsImRlYWRsaW5lIjoiMTE1NzkyMDg5MjM3MzE2MTk1NDIzNTcwOTg1MDA4Njg3OTA3ODUzMjY5OTg0NjY1NjQwNTY0MDM5NDU3NTg0MDA3OTEzMTI5NjM5OTM1In0sInRyYW5zZmVyRGV0YWlscyI6eyJ0byI6IjB4NDAwN0NFMjA4M2M3RjNFMTgwOTdhZUIzQTM5YmI4ZUMxNDlhMzQxZCIsInJlcXVlc3RlZEFtb3VudCI6IjI1MDAwMDAwMDAwMDAwMDAwMDAwIn0sIm93bmVyIjoiMHhmODdjYTQ1ODNDNzkyMjEyZTUyNzIwZDEyN0U3RTBBMzhCODE4YUQxIiwic2lnbmF0dXJlIjoiMHhlOTNkYjE0MjAwMzE0ODk4OWU0ZmU4OTc3MTEzY2M4YjdlNzc4NTljZjdjOWY2YmE5YzA3ZGY3ZTk2ZmUxNTMzMmY5MGIxNGM2YmU1YjVlMTMwMmVkZGQ1ZDE1NzMwM2U0MzNhZDI2MTA1MTExNWMwNGEwMTNiM2Y4NWJhZGVhYjFjIn0=)
> ```0x4007CE20...149a341d```

Multiplier of 0 still isn't working I guess? @0xcodercrane

**Comment by 0x4007** [comment]
>  Bounty hunters will spend more time deducting how to self assign an issue via a draft PR compared to the straight /assign command.

We currently have the capability for the bot to respond with the help/tutorial message whenever the user writes their first comment on any new repository. Within this tutorial message we can simply explain that in order to be assigned to the bounty now, they must open a draft pull request and associate it with the issue. 

> most of the issues are assigned via the /assign command.

The reason why I am not a fan of this approach is because I've seen a handful of instances where bounty hunters will reserve a task (disincentivizing anybody else from working on it) and then proceeding to not start on the task at all.

If they have a pull request opened, they at least got started with the task which I anecdotally observed as having a higher completion rate.

**Comment by rndquu** [comment]
> > Bounty hunters will spend more time deducting how to self assign an issue via a draft PR compared to the straight /assign command.
> 
> We currently have the capability for the bot to respond with the help/tutorial message whenever the user writes their first comment on any new repository. Within this tutorial message we can simply explain that in order to be assigned to the bounty now, they must open a draft pull request and associate it with the issue.
> 
> > most of the issues are assigned via the /assign command.
> 
> The reason why I am not a fan of this approach is because I've seen a handful of instances where bounty hunters will reserve a task (disincentivizing anybody else from working on it) and then proceeding to not start on the task at all.
> 
> If they have a pull request opened, they at least got started with the task which I anecdotally observed as having a higher completion rate.

This topic was actually discussed 2 months ago: https://t.me/c/1588400061/1/3176

```
TLDR; ask a great number of developers about the features they would want to use and then, based on the results, it will be clear what is better for them, the assign command or opening a PR, or perhaps both
```

**Comment by 0x4007** [comment]
I would like to try my approach because it solves the business problem of:

> If they have a pull request opened, they at least got started with the task which I anecdotally observed as having a higher completion rate.

This assumes that my anecdotal observation is accurate. I'm pretty confident in its accuracy because I am on GitHub monitoring these things pretty much daily.
